std/algorithm/comparison.d

```diff
@@ -1565,3 +1565,184 @@ unittest
+*/
+CommonType!Ts either(Ts...)(lazy Ts a)
+if (a.length >= 1 &&
+    !is(CommonType!Ts == void) &&
+    allSatisfy!(implicitlyConvertsToBool, Ts))
+{
+    foreach (e; a)
+        if (e)
+            return e;
+    return CommonType!Ts.init;
+}
+
+/// ditto
+auto ref either(Ts...)(ref Ts a)
+if (a.length >= 1 && allSameType!Ts &&
+    allSatisfy!(implicitlyConvertsToBool, Ts))
+{
+    foreach (ref e; a[0 .. $ - 1])
+        if (e)
+            return e;
+    return a[$ - 1];
+}
+
+///
+@safe pure unittest
+{
+    const a = 1;
+    const b = 2;
+    auto ab = either(a, b);
+    static assert(is(typeof(ab) == const(int)));
+    assert(ab == 1);
+
+    auto c = 1;
+    const d = 2;
+    auto cd = either(c, d);
+    static assert(is(typeof(cd) == int));
+    assert(cd == 1);
+
+    auto e = 0;
+    const f = 2;
+    auto ef = either(e, f);
+    static assert(is(typeof(ef) == int));
+    assert(ef == 2);
+
+    immutable p = 1;
+    immutable q = 2;
+    auto pq = either(p, q);
+    static assert(is(typeof(pq) == immutable(int)));
+    assert(pq == 1);
+
+    assert(either(3) == 3);
+    assert(either(3, 4) == 3);
+    assert(either(0, 4) == 4);
+    assert(either(0, 0) == 0);
+    assert(either("", "a") == "");
+
+    string r = null;
+    assert(either(r, "a") == "a");
+    assert(either("a", "") == "a");
+
+    immutable s = [1, 2];
+    assert(either(s) == s);
+
+    assert(either([0, 1], [1, 2]) == [0, 1]);
+    assert(either([0, 1], [1]) == [0, 1]);
+    assert(either("a", "b") == "a");
+
+    int x = 1;
+    int y = 2;
+    either(x, y) = 3;
+    assert(x == 3);
+    assert(y == 2);
+
+    static assert(!__traits(compiles, either(1, "a")));
+    static assert(!__traits(compiles, either(1.0, "a")));
+    static assert(!__traits(compiles, either('a', "a")));
+}
+
+/** Returns: Last Argument if all arguments implicitly bool-convert to $(D true)
+    otherwise $(D CommonType!T.init).
+
+    Similar to behaviour and operator in dynamic languages such as of Lisp's
+    (and a...) and Python's a and ....
+
+    TODO: Is inout Conversion!T the correct return value?
+*/
+CommonType!Ts every(Ts...)(lazy Ts a)
+if (Ts.length >= 1 &&
+    !is(CommonType!Ts == void) &&
+    allSatisfy!(implicitlyConvertsToBool, Ts))
+{
+    foreach (const ref e; a)
+        if (!e)
+            return CommonType!Ts.init;
+    return a[$ - 1];
+}
+
+///
+@safe pure unittest
+{
+    assert(every(3) == 3);
+
+    assert(every(3, 4) == 4);
+    assert(every(0, 4) == 0);
+    assert(every(0, 0) == 0);
+
+    assert(every([0, 1], [1, 2]) == [1, 2]);
+    assert(every([0, 1], [1]) == [1]);
+
+    assert(every("a", "b") == "b");
+    assert(every("", "b") == "b");
+
+    assert(every(cast(string)null, "b") == cast(string)null);
+
+    static assert(!__traits(compiles, every(1, "a")));
+    static assert(!__traits(compiles, every(1.0, "a")));
+    static assert(!__traits(compiles, every('a', "a")));
+}
+
+/** Evaluate all $(D parts) possibly digesting $(D whole).
```

Comment:
What does "to digest" mean in computer science? Only meaning I could think of was the checksum-related meaning.